src/System.Linq/tests/EnumerableTests.cs

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace System.Linq.Tests
+{
+    public partial class EnumerableTests
+    {
+        [Fact]
+        public void Average()
+        {
+            var array = Enumerable.Range(1, 10).ToArray();
```

Comment:
Well, there is no reason to go do ToArray() apart from avoiding to enumerate the enumerable multiple times. OK, I will modify the tests such that they test different input types.